driver/src/main/java/org/neo4j/driver/ResultTransformerProviders.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver;
+
+import static org.neo4j.driver.internal.CollectorResultTransformerProvider.EagerResultSummaryHandler;
+import static org.neo4j.driver.internal.CollectorResultTransformerProvider.PassThroughSummaryHandler;
+import static org.neo4j.driver.internal.CollectorResultTransformerProvider.RecordsCollectorProvider;
+
+import java.io.Serializable;
+import java.util.stream.Collector;
+import org.neo4j.driver.internal.CollectorResultTransformerProvider;
+import org.neo4j.driver.summary.ResultSummary;
+
+public class ResultTransformerProviders {
```

Comment:
`ResultTransformers` is a better name.

The methods would return `Supplier<?>`.